dask/dataframe/tests/test_categorical.py

```diff
@@ -266,6 +266,30 @@ def test_categorical_set_index(shuffle):
         assert list(sorted(d2.index.compute())) == ["b", "b", "c"]
 
 
+@pytest.mark.parametrize(
+    "ncategories", list(range(1, 6)), ids=lambda x: f"{x} categories"
+)
+@pytest.mark.parametrize(
+    "npartitions", list(range(1, 6)), ids=lambda x: f"{x} partitions"
+)
+def test_categorical_set_index_npartitions_vs_ncategories(npartitions, ncategories):
+    """https://github.com/dask/dask/issues/5343"""
+    rows_per_category = 10
+    n_rows = ncategories * rows_per_category
+
+    categories = [f"CAT{i}" for i in range(ncategories)]
+    pdf = pd.DataFrame(
+        {"id": categories * rows_per_category, "value": np.random.random(n_rows)}
+    )
+    ddf = dd.from_pandas(pdf, npartitions=npartitions)
+    ddf["id"] = ddf["id"].astype("category").cat.as_ordered()
+    ddf = ddf.set_index("id")
+    # Test passes if this worked and didn't raise any warnings
+
+    # NOTE: All observations of a category end up together, probably not desired:
```

Comment:
Wouldn't we expect this, given that `set_index` keeps similar values in the same partition?

Reply:
I'm not yet familiar enough with the internals, but from a user point of view I kind of expected the data to be balanced accross partitions. But if this is expected behavior then I'm glad I know now!